### ISSUE TITLE
Labwc: menu items in the bottom are cut out of the page when there are too many items

### ISSUE THREAD

**Issue by archisman-panigrahi** [raspberrypi/bookworm-feedback#313]
I have a lot of items in my Accessories, and they don't fit within the screen. In X11, I can scroll through the menu to access the items in the bottom.

In labwc, there is no way to scroll, and the menu items which do not fit inside the screen are not shown and there is no way to access them.

It would be nice to have a scrollbar in the menu in labwc.

Please let me know if I should report this bug in another repository.

**Comment by lurch** [comment]
I just spotted the same thing myself whilst investigating #337 
I asked @spl237 about it, and he said that the same menu-code works fine in X11, it just doesn't work in Wayland. He said that he's aware of the problem, but it's not currently at the top of his priority list.

I see that in #337 it looks like it's trying to create a single texture of 339x10745 pixels, so perhaps the reason that it's not showing the scroll-arrows on the menus is because it thinks that the "space available for the menu" isn't constrained at all, and so it's drawing the **entire** menu (even the off-screen parts), and that's why the menu isn't showing the scroll-arrows that you get on X11? :thinking: 

So sending a quick ping to @Consolatis in case he has any ideas / insight.

**Comment by Consolatis** [comment]
I assume we are talking about the panel menu, not a labwc built in menu here? If yes there is a weird thing with layershell that I've run into as well that makes xdg_popups not showing scroll bars when created via gtk layershell.
See
- https://github.com/wmww/gtk-layer-shell/issues/148

TLDR: Connecting to the `popped-up` signal and resizing the menu 'window' to `min(workarea, menu size)` should make the scrollbars appear correctly.

**Comment by spl237** [comment]
Now fixed - should be in apt later on today.
